js/src/components/google-combo-account-card/claim-ads-account/claim-ads-account.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Flex, FlexBlock } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ClaimAdsAccountButton from './claim-ads-account-button';
+import Section from '.~/wcdl/section';
+import useWindowFocus from '.~/hooks/useWindowFocus';
+import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
+import { useAppDispatch } from '.~/data';
+import './claim-ads-account.scss';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+
+/**
+ * ClaimAdsAccount component.
+ *
+ * @return {JSX.Element} ClaimAdsAccount component.
+ */
+const ClaimAdsAccount = () => {
+	const [ , forceUpdate ] = useState( 0 );
+	const isWindowFocused = useWindowFocus();
+	const { hasAccess, hasFinishedResolution, inviteLink, step } =
+		useGoogleAdsAccountStatus();
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+	const { invalidateResolution } = useAppDispatch();
+	const claimButtonClickedRef = useRef( false );
+	const claimInProgressRef = useRef( false );
+
+	useEffect( () => {
+		if ( isWindowFocused && claimButtonClickedRef.current ) {
+			invalidateResolution( 'getGoogleAdsAccountStatus' );
+			forceUpdate( ( prev ) => prev + 1 );
+		}
+	}, [ isWindowFocused, invalidateResolution ] );
+
+	useEffect( () => {
+		const handlePostClaimActions = async () => {
+			await upsertAdsAccount();
+			claimInProgressRef.current = false;
+		};
+
+		if (
+			hasAccess === true &&
+			step === 'conversion_action' &&
+			! claimInProgressRef.current
+		) {
+			claimInProgressRef.current = true;
+			handlePostClaimActions();
+		}
+
+		if ( claimButtonClickedRef.current && hasFinishedResolution ) {
+			claimButtonClickedRef.current = false;
+		}
+	}, [ hasAccess, step, upsertAdsAccount, hasFinishedResolution ] );
+
+	const isLoading =
+		loading || ( ! hasFinishedResolution && claimButtonClickedRef.current );
+
+	/**
+	 * Do not render anything if:
+	 *
+	 * 1. The user has access to the Google Ads account.
+	 * 2. The resolution has not finished and the claim button has not been clicked.
+	 */
+	if (
```

Comment:
@ankitrox I think the condition to render `ClaimAdsAccount` should be in `ConnectedGoogleComboAccountCard`
and we can check as follows:
```js
const shouldClaimGoogleAdsAccount = Boolean(
		googleAdsAccount.id && hasAccess === false
	);
```

Thus we won't need to track for any button click. Side note, we don't need to check for `! hasFinishedResolution` since `hasAccess` will be falsy if that's not the case.